Userland/Libraries/LibWeb/Fetch/Body.cpp

```diff
@@ -135,9 +136,13 @@ WebIDL::ExceptionOr<JS::Value> package_data(JS::Realm& realm, ByteBuffer bytes,
     case PackageDataType::JSON:
         // Return the result of running parse JSON from bytes on bytes.
         return Infra::parse_json_bytes_to_javascript_value(realm, bytes);
-    case PackageDataType::Text:
+    case PackageDataType::Text: {
         // Return the result of running UTF-8 decode on bytes.
-        return JS::PrimitiveString::create(vm, TRY_OR_THROW_OOM(vm, String::from_utf8(bytes)));
+        auto decoder = TextCodec::decoder_for("UTF-8"sv);
+        VERIFY(decoder.has_value());
+
+        return JS::PrimitiveString::create(vm, TextCodec::convert_input_to_utf8_using_given_decoder_unless_there_is_a_byte_order_mark(*decoder, bytes).release_value_but_fixme_should_propagate_errors());
```

Comment:
shouldn't we be able to propagate the errors from `convert_input_to_utf8_using_given_decoder_unless_there_is_a_byte_order_mark` ?

Reply:
I couldn't make it work, it complained about JS:Value's constructor when I used TRY. But I'm not sure if that was a correct way of doing it

Reply:
Use `TRY_OR_THROW_OOM`:
```c++
auto utf8_text = TRY_OR_THROW_OOM(vm, TextCodec::convert_input_to_utf8_using_given_decoder_unless_there_is_a_byte_order_mark(*decoder, bytes));
return JS::PrimitiveString::create(vm, move(utf8_text));
```

`TextCodec::convert_input_to_utf8...` returns an `AK::Error`, and only when we run out-of-memory. We use `TRY_OR_THROW_OOM` to convert such `Error` instances to a `JS::Completion`.